lib/3scale/api/client.rb

```diff
@@ -51,9 +51,8 @@ def show_application(id)
       # @return [Hash]
       # @param [Fixnum] id Application ID
       # @param [String] user_key Application User Key
-      # @param [String] application_id Application App ID
-      def find_application(id: nil, user_key: nil, application_id: nil)
-        params = { application_id: id, user_key: user_key, app_id: application_id }.reject { |_, value| value.nil? }
+      def find_application(id: nil, user_key: nil)
```

Comment:
Why removing the functionality?

Reply:
3scale API does not allow to find app by `application_id`. It is provided by `id` param.

Reply:
I believe it does:
https://github.com/3scale/porta/blob/704d2061af6689fb43647e19d060017842418bbc/app/controllers/admin/api/applications_controller.rb#L70-L83

Reply:
Tests say it does not:

```ruby                                
res = client.find_application(application_id: 1409618277030)                                
```

you get `#<Net::HTTPNotFound:0x0000564952bd2be8> (ThreeScale::API::HttpClient::NotFoundError)`

On the other hand,

```ruby                                
res = client.find_application(id: 1409618277030)                                
```

then you get the application object.

You can try yourself. That's why the integration test was in pending state.

Reply:
And the application/service is indeed configured to be app_id+app_key, not user_key?
Because that is what the source code I linked in 
https://github.com/3scale/3scale-api-ruby/pull/51#discussion_r264368923 checks.

Reply:
I see now what's the issue. I confused application `id` with `app_id` for services with authentication type `app_id, app_key`.

thnx for the clarification  @mikz 

Will update the test to create app on a service with auth `[app_id, app_key]` and check `find_by_app_id` works as expected